src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-details/osd-details.component.ts

```diff
@@ -35,8 +35,11 @@ export class OsdDetailsComponent implements OnChanges {
     this.osdService.getDetails(this.osd.tree.id).subscribe((data: any) => {
       this.osd.details = data;
       if (!_.isObject(data.histogram)) {
-        this.osd.histogram_failed = data.histogram;
+        this.osd.histogram_succeeded = false;
```

Comment:
As you are only using `histogram_succeeded` once and inverted you can remove the `else` part and also this variable. It should be enough to only set `histogram_error_msg` once `data.histogram` is not an object. If it's not set it will be `undefined` if you try to access it.